File: letter_grade.clj
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
;;;; Name: Lily McAboy
;;;; Date: 2/5/2024
;;;; CS 214 Project 03
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;

(ns letter_grade) ; namespace function names the program

(defn letterGrade [usr_grade]
(let [grade (/ usr_grade 10)]
 (cond
  (= grade 10) "A"
  (= grade 8) "B"
  (= grade 7) "C"
  (= grade 6) "D"
  :else "F"
 ))
 )

(defn -main []
 (println "\nEnter your grade:,")
 (let
  [ grade (read) ]

  (assert (>= radius 0) "-main: grade must be positive")
  (printf "\nYour letter grade is %f\n\n" (letterGrade grade))
 )
 )

File: letter_grade.rb
# Name: Lily McAboy
# Date: 02/5/2024
###############################################################

def letter_grade(grade)
    usr_grade = grade / 10
    case usr_grade
    when 10 || 9
        "A"
    when 8
        "B"
    when 7
        "C"
    when 6 
        "D"
    else
        "F"
end

if __FILE__ == $0
   puts "Enter your grade: "
   grade = gets.chomp.to_f
   print "Your letter grade is: "
   puts letter_grade(grade)
end